eZ/Bundle/EzPublishCoreBundle/Command/RegenerateUrlAliasesCommand.php

```diff
@@ -0,0 +1,241 @@
+            }
+        );
+
+        $helper = $this->getHelper('question');
+        $question = new ConfirmationQuestion(
+            sprintf(
+                '<info>Found %d Locations. Do you want to proceed? [y/N] </info>',
+                $locationsCount
+            ),
+            false
+        );
+        if (!$helper->ask($input, $output, $question)) {
+            return;
+        }
+
+        $output->writeln('Regenerating System URL aliases...');
+
+        $progressBar = $this->getProgressBar($locationsCount, $output);
+        $progressBar->start();
+
+        for ($offset = 0; $offset <= $locationsCount; $offset += $iterationCount) {
+            $locations = $this->repository->sudo(
+                function (Repository $repository) use ($offset, $iterationCount) {
+                    return $repository->getLocationService()->loadAllLocations($offset, $iterationCount);
+                }
+            );
+            $this->processLocations($locations, $progressBar);
+            // force garbage collection
+            gc_collect_cycles();
+        }
+        $progressBar->finish();
+        $output->writeln('');
+
+        $output->writeln('<info>Cleaning up corrupted URL aliases...</info>');
+        $corruptedAliasesCount = $this->repository->sudo(
+            function (Repository $repository) {
+                return $repository->getURLAliasService()->deleteCorruptedUrlAliases();
+            }
+        );
+        $output->writeln("<info>Done. Deleted {$corruptedAliasesCount} entries.</info>");
+        $output->writeln('Purging HTTP cache...');
+        $this->cachePurger->purgeAll();
+        $output->writeln('<info>Done.</info>');
+    }
+
+    /**
+     * Return configured progress bar helper.
+     *
+     * @param int $maxSteps
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     *
+     * @return \Symfony\Component\Console\Helper\ProgressBar
+     */
+    protected function getProgressBar($maxSteps, OutputInterface $output)
+    {
+        $progressBar = new ProgressBar($output, $maxSteps);
+        $progressBar->setFormat(
+            ' %current%/%max% [%bar%] %percent:3s%% %elapsed:6s%/%estimated:-6s% %memory:6s%'
+        );
+
+        return $progressBar;
+    }
+
+    /**
+     * Process single results page of fetched Locations.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Location[] $locations
+     * @param \Symfony\Component\Console\Helper\ProgressBar $progressBar
+     */
+    private function processLocations(array $locations, ProgressBar $progressBar)
+    {
+        // get published only ContentInfo list
+        $contentInfoList = array_filter(
+            array_map(
+                function (Location $location) {
+                    return $location->contentInfo;
+                },
+                $locations
+            ),
+            function (ContentInfo $contentInfo) {
+                return $contentInfo->isPublished();
```

Comment:
Is this to prepare for location drafts feature? Afaik only published content will be returned when we query via locations. Or did you get issues before you added this and filter below for published flag?

Reply:
No, I had in mind drafts with assigned location but not published yet. However indeed `ezcontentobject_tree` should have only published ones, my mistake. Removed this check also via 01bb253.